src/lib.rs

```diff
@@ -85,12 +85,16 @@ fn get_class(conn: &xcb::Connection, id: u32, options: &Options) -> Result<Strin
 
     let result = String::from_utf8(buf)?;
     let mut results = result.split('\0');
-    results.next_back();
+    let wm_instance = results.next().unwrap();
     let mut results_with_icons = results.map(|class| {
         let class_display_name = match options.aliases.get(class) {
             Some(alias) => alias,
             None => class,
         };
+        let mut class = class;
```

Comment:
Here you just check whether the icon matches in config, but the `class_display_name` still uses WM_CLASS to look up aliases.

Reply:
Right, I didn't notice this during testing because I only display icons myself.